Added basic toggle - cabled mode for devices

Meister1593 opened the pull request:
Appears for each device, and when toggled - automatically adds local ip, disables client discovery and switches to TCP.
When disabled, it removes local ip (though really not that necessary, still did it), enables client discovery, and leaves TCP as is (because alvr doesn't know about previous state, it's safer to just leave it as is).

Supposed to be very easy and useful toggle to not force people to go and learn where those settings are.

In addition, i made label a hyperlink so people can check out what they need to be able to use that toggle.

I think that's my first "real" pr so any comments about how to do things better (especially when writing settings) appreciated.

![image](https://github.com/alvr-org/ALVR/assets/7141787/8f517c31-4890-4564-aa3d-c4290e929500)

l33tlinuxh4x0r commented:
I would change the wording from Cabled to "Use USB" or "Use Cable" for the toggle switch.

Meister1593 commented:
Also noticed this, unsure why spacing doesnt get enforced for all elements, only for last one
![image](https://github.com/alvr-org/ALVR/assets/7141787/ff2bb922-31fc-4226-9f03-9b36fe501e9e)

Meister1593 commented:
> I would change the wording from Cabled to "Use USB" or "Use Cable" for the toggle switch.

done, changed to `Use Cable`

Meister1593 commented:
I also tried going into presets area (creating single toggle for multiple paths) and doing some things with them but i got too deep with creating boolean option within it and in the end it didn't work unfortunately

l33tlinuxh4x0r commented:
> Also noticed this, unsure why spacing doesnt get enforced for all elements, only for last one ![image](https://user-images.githubusercontent.com/7141787/278842092-ff2bb922-31fc-4226-9f03-9b36fe501e9e.png)

I think that I found the issue... check line 118 of file /alvr/dashboard/src/dashboard/components/connections.rs
`Grid::new(2).num_columns(2).show(ui, |ui| {`
try changing it to 
`Grid::new(2).num_columns(3).show(ui, |ui| {`

just a guess as I haven't compiled it yet. I will reboot to linux and try to compile it and see.

l33tlinuxh4x0r commented:
Ignore last post, it breaks things. I tested and found a better fix. This should probably be implemented anyway so that you can always see everything. I set the minimum window size to the default window size so you can make the window bigger but not smaller. I changed /alvr/dashboard/src/main.rs after line 91 `initial_window_size: Some(egui::vec2(870.0, 600.0)),`
I added:
```rust
min_window_size: Some(egui::vec2(870.0, 600.0)),
```

Meister1593 commented:
> Ignore last post, it breaks things. I tested and found a better fix. This should probably be implemented anyway so that you can always see everything. I set the minimum window size to the default window size so you can make the window bigger but not smaller. I changed /alvr/dashboard/src/main.rs after line 91 `initial_window_size: Some(egui::vec2(870.0, 600.0)),` I added:
> 
> ```rust
> min_window_size: Some(egui::vec2(870.0, 600.0)),
> ```

it works, though not sure if hi-dpi users (steam deck users, for example) will like it

l33tlinuxh4x0r commented:
> > Ignore last post, it breaks things. I tested and found a better fix. This should probably be implemented anyway so that you can always see everything. I set the minimum window size to the default window size so you can make the window bigger but not smaller. I changed /alvr/dashboard/src/main.rs after line 91 `initial_window_size: Some(egui::vec2(870.0, 600.0)),` I added:
> > ```rust
> > min_window_size: Some(egui::vec2(870.0, 600.0)),
> > ```
> 
> it works, though not sure if hi-dpi users (steam deck users, for example) will like it

For what it is worth I tried making elements specific sizes too but the main window ignores this. That is the only thing that I could see that worked. Does anyone have a steam deck that can post a screenshot of how big the window looks on it?

zarik5 commented:
I believe we should arrange elements (labels/buttons) on two rows per client.

zarik5 commented:
Another option is to put the switch inside the Edit client dialog box.

zarik5 commented on alvr/packets/src/lib.rs:
```diff
@@ -188,6 +188,7 @@ pub enum ClientListAction {
     RemoveEntry,
     UpdateCurrentIp(Option<IpAddr>),
     SetConnectionState(ConnectionState),
+    ToggleCabled,
```
I believe this variant should take a bool. It's because the behavior might get confusing if there are two dashboards concurrently calling this.

Meister1593 commented:
![image](https://github.com/alvr-org/ALVR/assets/7141787/8142f148-919f-4e95-b14f-3c37b4bbfc70)
Finalized UI look

l33tlinuxh4x0r commented:
> ![image](https://user-images.githubusercontent.com/7141787/279615829-8142f148-919f-4e95-b14f-3c37b4bbfc70.png) Finalized UI look

Looks great!

zarik5 commented:
For symmetry i believe we should also put a frame around the new clients individually.

Meister1593 commented:
> For symmetry i believe we should also put a frame around the new clients individually.

You mean the ones that you need to trust first? That could be arranged, yes, i forgot about them

Meister1593 commented:
![image](https://github.com/alvr-org/ALVR/assets/7141787/2d3629e7-c10b-4726-9708-60b13ec8525e)

zarik5 commented:
Looking awesome! 😄

l33tlinuxh4x0r commented:
Agreed.

> Looking awesome! 😄
